### PR TITLE
Added Accessibility Support for Screenreaders

### PR DESCRIPTION
**Pull request by hannesa2**
It is a copy of https://github.com/PhilJay/MPAndroidChart/pull/4936

### CODE REVIEW

**Comment by hannesa2** [comment]
Who needs this ?

**Comment by Karkoon** [comment]
I'm not personally in need of this, but I wanted to mention that in 2025, in the EU, all commercial apps will have to support accessibility in a more organized manner.  https://ec.europa.eu/social/main.jsp?catId=1202&intPageId=5581&langId=en

I didn't research it that much to be honest, but it might be an important feature.

**Comment by hannesa2** [comment]
It would be nice, if @ish-joshi agree to this pull request.

Independent of the up coming EU law, it's an issue of commercial apps. If they need it, they can make pull request, and I'll merge it. Maybe this pull request will solve this in advance

File: MPChartLib/src/main/java/com/github/mikephil/charting/charts/CandleStickChart.java [hunk 1]
```diff
@@ -41,4 +41,9 @@ protected void init() {
     public CandleData getCandleData() {
         return mData;
     }
+
+    @Override
+    public String getAccessibilityDescription() {
+        return "This is a candlestick";
```
Comment (dan-ryan) [inline MPChartLib/src/main/java/com/github/mikephil/charting/charts/CandleStickChart.java:47]: To be consistent, shouldn't this say, "This is a candlestick chart"?

Comment (hannesa2) [reply MPChartLib/src/main/java/com/github/mikephil/charting/charts/CandleStickChart.java]: Sure, that is easy

**Comment by ish-joshi** [comment]
I'm happy for anyone to take the lead, initially this was started as a way to get the conversation going. 

Whatever be next steps in making this widespread and more useful for users, I'm happy to provide support there.

**Pull request merged**